symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -178,11 +181,22 @@ public V4Message send(@Nonnull V4Stream stream, @Nonnull String message) {
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
-  public V4Message send(@Nonnull V4Stream stream, @Nonnull String template, Object parameters)
-      throws TemplateException {
+  public V4Message sendWithTemplate(@Nonnull V4Stream stream, @Nonnull String template, Object parameters) {
     return send(stream.getStreamId(), templateResolver.resolve(template).process(parameters));
   }
 
+  /**
+   * Sends a templated to the stream ID of the passed {@link V4Stream} object.
+   *
+   * @param stream     the stream to send the message to
+   * @param template   the template name to be used to produce the message
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @throws TemplateException
+   */
+  public V4Message sendWithTemplate(@Nonnull V4Stream stream, @Nonnull String template) {
```

Comment:
Changing name of this method will also break `generator-symphony`.

Reply:
Do we plan to introduce a message object (to avoid passing many parameters and having many methods to send a message)? If so we will probably go back to a single send method no?

Reply:
@symphony-youri yes I think it's a better solution but we can consider it next sprint?

Reply:
Yes I'm just pointing that we might change the API here, update the doc and generator for it and then do it again next sprint so maybe it is easier to do it this sprint (how given how fast you are I'm pretty sure it could be done:))

Reply:
We have a lot of things already plan for next sprint, could you tackle this idea as part of this PR? As it would also really reduce the complexity of this class.